wrangler-service/src/main/java/co/cask/wrangler/service/hiveserver2/Hiveserver2Service.java

```diff
@@ -398,41 +401,56 @@ public void specification(HttpServiceRequest request, final HttpServiceResponder
    private JsonObject getHiveTableSchema(HttpServiceRequest request, String connectionURL, String dbName, String tableName){
      JsonObject schema = null;
 
-     java.sql.Connection hiveConnection=null;
-     PreparedStatement ps = null;
-     ResultSet rs = null;
-     try {
-       String updatedConnectionURL = addDelegationTokenUpdateURL(connectionURL, request);
-       hiveConnection = DriverManager.getConnection(updatedConnectionURL);
-
-       ps = hiveConnection.prepareStatement(String.format("describe %s.%s", dbName, tableName));
-       rs = ps.executeQuery();
-
-       schema = new JsonObject();
-       schema.addProperty("type", "record");
-       schema.addProperty("name", "etlSchemaBody");
-
-       JsonArray fields = new JsonArray();
-       while (rs.next()) {
-         JsonObject column = new JsonObject();
-         column.addProperty("name", rs.getString("col_name"));
-         column.addProperty("type", rs.getString("data_type"));
-         fields.add(column);
+     if(getContext().getRuntimeArguments().containsKey(HIVE_ENABLE_ADDING_SCHEMA)
+             && getContext().getRuntimeArguments().get(HIVE_ENABLE_ADDING_SCHEMA).equalsIgnoreCase("true")) {
+
+       java.sql.Connection hiveConnection = null;
+       PreparedStatement ps = null;
+       ResultSet rs = null;
+       try {
+         String updatedConnectionURL = addDelegationTokenUpdateURL(connectionURL, request);
+         hiveConnection = DriverManager.getConnection(updatedConnectionURL);
+
+         ps = hiveConnection.prepareStatement(String.format("describe %s.%s", dbName, tableName));
+         rs = ps.executeQuery();
+
+         schema = new JsonObject();
+         schema.addProperty("type", "record");
+         schema.addProperty("name", "etlSchemaBody");
+
+         JsonArray fields = new JsonArray();
+         while (rs.next()) {
+           JsonObject column = new JsonObject();
+           column.addProperty("name", rs.getString("col_name"));
+           column.addProperty("type", getCDAPDataType(rs.getString("data_type")));
+           fields.add(column);
+         }
+         schema.add("fields", fields);
+       } catch (Exception e) {
+         LOG.error("Error fetching hive table schema ", e);
+         e.printStackTrace();
```

Comment:
Rather than dumping to stdout, dump stack trace to logger

Reply:
LOG.error will take care of it, removed `e.printStackTrace()`